File: HomeTask/app/src/main/java/com/example/hometask/Client_Fragment.java
package com.example.hometask;

import android.content.Context;
import android.content.Intent;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.j256.ormlite.android.apptools.OpenHelperManager;
import com.j256.ormlite.dao.RuntimeExceptionDao;

import java.util.List;


public class Client_Fragment extends Fragment {
    DataBase dataBase;


    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        dataBase = OpenHelperManager.getHelper(getContext(), DataBase.class);

        RecyclerView productRecycler = (RecyclerView) inflater.inflate(R.layout.fragment_client, container, false);

        RuntimeExceptionDao<Client, Integer> runtimeExceptionDao = dataBase.getRuntimeExceptionDao();
        List<Client> clientList = runtimeExceptionDao.queryForAll();

        String[] surname = new String[clientList.size()];

        for (int i=0; i<clientList.size(); i++) {
            surname[i] = clientList.get(i).getSurname();
        }

        int[] id = new int[clientList.size()];

        for (int i=0; i<clientList.size(); i++) {
            id[i] = clientList.get(i).getId();
            Log.d("ARRAYID", String.valueOf(id[i]));
        }

        String[] name = new String[clientList.size()];

        for (int i=0; i<clientList.size(); i++) {
            name[i] = clientList.get(i).getName();
        }

        String[] patronymic = new String[clientList.size()];

        for (int i=0; i<clientList.size(); i++) {
            patronymic[i] = clientList.get(i).getPatronymic();
        }

        String[] position = new String[clientList.size()];

        for (int i=0; i<clientList.size(); i++) {
            position[i] = clientList.get(i).getPosition();
        }

        String[] date = new String[clientList.size()];

        for (int i=0; i<clientList.size(); i++) {
            date[i] = clientList.get(i).getDate();
        }

            Adapter adapter = new Adapter(surname, name, patronymic, position, date, id);
            productRecycler.setAdapter(adapter);
            LinearLayoutManager layoutManager = new LinearLayoutManager(getActivity());
            productRecycler.setLayoutManager(layoutManager);

            adapter.setListener(new Adapter.Listener() {
                @Override
                public void onClick(int i) {
                    Intent intent = new Intent(getActivity(), AddActivity.class);
                    intent.putExtra(AddActivity.EXTRA_CLIENT_ID, i);
                    getActivity().startActivity(intent);
                }
            });
        return productRecycler;

    }
}

File: HomeTask/app/src/main/java/com/example/hometask/Client_Fragment.kt
package com.example.hometask

import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.j256.ormlite.android.apptools.OpenHelperManager


class Client_Fragment : Fragment() {
    var dataBase: DataBase? = null


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {

        dataBase = OpenHelperManager.getHelper(context, DataBase::class.java)

        val productRecycler = inflater.inflate(R.layout.fragment_client, container, false) as RecyclerView

        val runtimeExceptionDao = dataBase?.getRuntimeExceptionDao()
        val clientList = runtimeExceptionDao!!.queryForAll()

        val surname = arrayOfNulls<String>(clientList.size)

        for (i in clientList.indices) {
            surname[i] = clientList[i].surname
        }

        val id = IntArray(clientList.size)

        for (i in clientList.indices) {
            id[i] = clientList[i].id
            Log.d("ARRAYID", id[i].toString())
        }

        val name = arrayOfNulls<String>(clientList.size)

        for (i in clientList.indices) {
            name[i] = clientList[i].name
        }

        val patronymic = arrayOfNulls<String>(clientList.size)

        for (i in clientList.indices) {
            patronymic[i] = clientList[i].patronymic
        }

        val position = arrayOfNulls<String>(clientList.size)

        for (i in clientList.indices) {
            position[i] = clientList[i].position
        }

        val date = arrayOfNulls<String>(clientList.size)

        for (i in clientList.indices) {
            date[i] = clientList[i].date
        }

        val adapter = Adapter(surname, name, patronymic, position, date, id)
        productRecycler.adapter = adapter
        val layoutManager = LinearLayoutManager(activity)
        productRecycler.layoutManager = layoutManager

        adapter.setListener(object : Adapter.Listener {
            override fun onClick(i: Int) {
                val intent = Intent(activity, AddActivity::class.java)
                intent.putExtra(AddActivity.EXTRA_CLIENT_ID, i)
                activity!!.startActivity(intent)
            }
        })
        return productRecycler

    }
}
